devices/Ssd13xx/Ssd13xx.cs

```diff
@@ -101,5 +142,172 @@ protected SpanByte SliceGenericBuffer(int start, int length)
+                    var b = bitmap[(yO * width) + xA];
+                    byte mask = 0x01;
+                    for (var pixel = 0; pixel < 8; pixel++)
+                    {
+                        DrawPixel(x + (8 * xA) + pixel, y + yO, (b & mask) > 0);
+                        mask <<= 1;
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="x"></param>
+        /// <param name="y"></param>
+        /// <param name="str"></param>
+        public void DrawString(int x, int y, string str)
+        {
+            byte[] bitMap = GetTextBytes(str);
+
+            DrawBitmap(x, y, bitMap.Length / Font.Height, Font.Height, bitMap);
+        }
+
+
+        /// <summary>
+        /// Get the bytes to be drawn on the screen for text, from the font
+        /// </summary>
+        /// <param name="text">Strint to be shown on the screen.</param>
+        /// <returns></returns>
+        byte[] GetTextBytes(string text)
+        {
+            byte[] bitMap;
+
+            if (Font.Width == 8)
+            {
+                bitMap = new byte[text.Length * Font.Height * Font.Width / 8];
+
+                for (int i = 0; i < text.Length; i++)
+                {
+                    var characterMap = Font[text[i]];
+
+                    for (int segment = 0; segment < Font.Height; segment++)
+                    {
+                        bitMap[i + (segment * text.Length)] = characterMap[segment];
+                    }
+                }
+            }
+            else
+            {
+                throw new Exception("Font width must be 8");
+            }
+
+            return bitMap;
+        }
+
+        /// <summary>
+        /// Displays the information on the screen.
+        /// </summary>
+        public void Display()
+        {
+            for (byte i = 0; i < Pages; i++)
+            {
+                _pageCmd[1] = (byte)(PageAddress.Page0 + i); // page number
+                _i2cDevice.Write(_pageCmd);
+
+                _pageData[0] = 0x40; // is data
+                Array.Copy(_genericBuffer, i * Width, _pageData, 1, Width);
+                _i2cDevice.Write(_pageData);
+            }
+        }
+
+        /// <summary>
+        /// Clears the screen.
+        /// </summary>
+        public void ClearScreen()
+        {
+            Array.Clear(_genericBuffer, 0, _genericBuffer.Length);
+
+            Display();
+        }
+
+
+
+        /// <summary>
+        ///     Sequence of bytes that should be sent to a 128x64 OLED display to setup the device.
+        ///		First byte is the command byte 0x00.
+        /// </summary>
+        private readonly byte[] _oled128x64Init =
+        {
+            0x00, 0xae, 0xd5, 0x80, 0xa8, 0x3f, 0xd3, 0x00, 0x40 | 0x0, 0x8d, 0x14, 0x20, 0x00, 0xa0 | 0x1, 0xc8,
+            0xda, 0x12, 0x81, 0xcf, 0xd9, 0xf1, 0xdb, 0x40, 0xa4, 0xa6, 0xaf
+        };
+
+        /// <summary>
```

Comment:
please adjust the comments by removing the extra spaces
And also a space alignement problem. In Visual Studio, you can use ctrl + k + d
This automatically correct those kind of miss alignements.

Reply:
@gligorov this is really perfect!!! Thanks a lot!